rust/connlib/tunnel/src/lib.rs

```diff
@@ -111,6 +117,8 @@ impl ClientTunnel {
 
     pub fn poll_next_event(&mut self, cx: &mut Context<'_>) -> Poll<Result<ClientEvent, Error>> {
         for _ in 0..MAX_EVENTLOOP_ITERS {
+            self.role_state.set_ip_stack(self.io.ip_stack()?);
```

Comment:
This can only change after sending right? can we set it there?

Reply:
After sending and after roaming. I am a bit worried that setting it conditionally on something might make us miss a spot later.

Are you worried about performance?